src/Futhark/CLI/CompareBench.hs

```diff
@@ -0,0 +1,322 @@
+-- | @futhark compare-bench@
+module Futhark.CLI.CompareBench (main) where
+
+import Control.Exception (catch)
+import Data.Bifunctor (Bifunctor (bimap, second))
+import Data.ByteString.Lazy.Char8 qualified as LBS
+import Data.Either qualified as E
+import Data.List qualified as L
+import Data.Map qualified as M
+import Data.Text qualified as T
+import Data.Vector qualified as V
+import Futhark.Bench
+import Futhark.Util.Options (mainWithOptions)
+import Statistics.Sample qualified as S
+import System.Console.ANSI (hSupportsANSI)
+import System.IO (stdout)
+import Text.Printf (printf)
+
+-- | Record that summerizes a comparison between two benchmarks.
+data SpeedUp = SpeedUp
+  { -- | What factor the benchmark is improved by.
+    speedup :: Double,
+    -- | Memory usage.
+    memoryUsage :: M.Map T.Text Double,
+    -- | If the speedup was significant.
+    significant :: Bool
+  }
+  deriving (Show)
+
+-- | Terminal colors used when printing the comparisons. Some of these are not
+-- colors ways of emphasising text.
+data Colors = Colors
+  { -- | The header color.
+    header :: String,
```

Comment:
I doubt it matters in practice for a program this small, but I/we have a general principle of never using `String`.  Use the type `Data.Text.Text` instead.  (The fact that the default String type in Haskell is really bad is an unfortunate historical accident).  The main difference is that you'll use `<>` for concatenation instead of `++`.